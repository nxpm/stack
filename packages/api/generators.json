{
  "$schema": "http://json-schema.org/schema",
  "name": "api",
  "version": "0.0.1",
  "generators": {
    "api": {
      "factory": "./src/generators/api/generator-api#generatorApi",
      "schema": "./src/generators/api/schema.json",
      "description": "api generator"
    },
    "api-e2e": {
      "factory": "./src/generators/api-e2e/generator-api-e2e#generatorApiE2e",
      "schema": "./src/generators/api-e2e/schema.json",
      "description": "api-e2e generator"
    },
    "api-feature": {
      "factory": "./src/generators/api-feature/generator-api-feature#generatorApiFeature",
      "schema": "./src/generators/api-feature/schema.json",
      "description": "api-feature generator"
    },
    "api-lib": {
      "factory": "./src/generators/api-lib/generator-api-lib#generatorApiLib",
      "schema": "./src/generators/api-lib/schema.json",
      "description": "api-lib generator"
    }
  }
}
