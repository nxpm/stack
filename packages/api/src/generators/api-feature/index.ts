export * from './generator-api-feature-account'
export * from './generator-api-feature-auth'
export * from './generator-api-feature-core'
export * from './generator-api-feature-user'
