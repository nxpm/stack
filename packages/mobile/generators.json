{
  "$schema": "http://json-schema.org/schema",
  "name": "mobile",
  "version": "0.0.1",
  "generators": {
    "mobile": {
      "factory": "./src/generators/mobile/generator-mobile#generatorMobile",
      "schema": "./src/generators/mobile/schema.json",
      "description": "mobile generator"
    }
  }
}
