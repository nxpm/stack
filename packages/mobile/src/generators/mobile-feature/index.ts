export * from './create-mobile-feature'
export * from './generate-mobile-feature-about'
export * from './generate-mobile-feature-account'
export * from './generate-mobile-feature-auth'
export * from './generate-mobile-feature-core'
export * from './generate-mobile-feature-dashboard'
export * from './generate-mobile-feature-layout'
export * from './generate-mobile-feature-shell'
export * from './generator-mobile-feature'
