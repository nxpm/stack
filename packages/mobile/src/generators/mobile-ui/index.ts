export * from './generate-mobile-ui-lib-button'
export * from './generate-mobile-ui-lib-form'
export * from './generate-mobile-ui-lib-icon'
export * from './generate-mobile-ui-lib-loader'
export * from './generate-mobile-ui-lib-page'
export * from './generate-mobile-ui-lib-page-header'
export * from './generate-mobile-ui-lib-sidebar-page'
export * from './generate-mobile-ui-lib-table'
