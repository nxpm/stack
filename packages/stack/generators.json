{
  "$schema": "http://json-schema.org/schema",
  "name": "stack",
  "version": "0.0.1",
  "generators": {
    "stack": {
      "factory": "./src/generators/stack/generator",
      "schema": "./src/generators/stack/schema.json",
      "description": "stack generator"
    }
  }
}
