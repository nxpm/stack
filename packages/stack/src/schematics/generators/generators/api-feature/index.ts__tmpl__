import { formatFiles, Tree } from '@nrwl/devkit'
import { libraryGenerator } from '@nrwl/nest'

async function createLibrary(host: Tree, appName, name, dryRun = false) {
  const directory = `${appName}/${name}`
  await libraryGenerator(host, { name: 'data-access', directory, tags: `scope:${appName},type:data-access`, dryRun })
  await libraryGenerator(host, { name: 'feature', directory, tags: `scope:${appName},type:feature`, dryRun })
}

export default async function (host: Tree, schema: { name: string; appName: string; dryRun?: boolean }) {
  const appName = schema.appName || 'api'
  await createLibrary(host, appName, schema.name, schema.dryRun)
  await formatFiles(host)
}
