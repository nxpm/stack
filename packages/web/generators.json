{
  "$schema": "http://json-schema.org/schema",
  "name": "web",
  "version": "0.0.1",
  "generators": {
    "web": {
      "factory": "./src/generators/web/generator-web#generatorWeb",
      "schema": "./src/generators/web/schema.json",
      "description": "web generator"
    }
  }
}
