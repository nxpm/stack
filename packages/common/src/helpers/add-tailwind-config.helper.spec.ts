import { readProjectConfiguration, Tree } from '@nrwl/devkit'
import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing'
import { addTailwindConfig } from './add-tailwind-config.helper'
import { applicationGenerator } from '@nrwl/angular/generators'

describe('add-tailwind-config helper', () => {
  let appTree: Tree
  const options = { name: 'app-name-test' }
  beforeEach(() => {
    appTree = createTreeWithEmptyWorkspace()
  })

  it('should add Tailwind Config', async () => {
    await applicationGenerator(appTree, options)
    addTailwindConfig(appTree, options.name)
    const app = readProjectConfiguration(appTree, options.name)

    expect(appTree.read(`${app.sourceRoot}/index.html`).toString()).toMatchSnapshot()
    expect(appTree.read('tailwind.config.js').toString()).toMatchSnapshot()
  })
})
