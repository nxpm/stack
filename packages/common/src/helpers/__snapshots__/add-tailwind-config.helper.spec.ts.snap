// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add-tailwind-config helper should add Tailwind Config 1`] = `
"<!doctype html>
<html lang=\\"en\\">
<head>
  <meta charset=\\"utf-8\\">
  <title>AppNameTest</title>
  <base href=\\"/\\">
  <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
  <link rel=\\"icon\\" type=\\"image/x-icon\\" href=\\"favicon.ico\\">
</head>
<body class=\\"dark\\">
  <proj-root></proj-root>
</body>
</html>
"
`;

exports[`add-tailwind-config helper should add Tailwind Config 2`] = `
"module.exports = {
      mode: process.env.TAILWIND_MODE ? 'jit' : '',
      purge: ['./apps/**/*.{html,ts}', './libs/**/*.{html,ts}'],
      darkMode: 'class', // or 'media' or 'class'
      theme: {
        extend: {},
      },
      variants: {
        extend: {
          opacity: ['disabled'],
          cursor: ['disabled'],
        },
      },
      plugins: [require('@tailwindcss/forms')],
    }"
`;
